NumRush.py

```diff
@@ -364,21 +367,38 @@ def run(self):
             pygame.display.update()
             fpsClock.tick(FPS)
 
-    def restore_game(self, high_score=0, current_score=0, game=0):
+    def restore_game(self, high_score=0, current_score=0):
         ''' Restore a game from the Journal '''
-        # TODO: Save/restore recall list for game 2
         hscore = high_score
         score = current_score
-        self.game = game
-        for i, dot in enumerate(dot_list):
-            self._dots[i].type = dot
-            if dot == -1:
-                self._dots[i].hide()
-        self._new_game(restore=True)
 
     def save_game(self):
         ''' Return game state for saving to Journal '''
-        return  hscore, score, self.game
+        return  hscore, score
+
+    def read_highscore(self):
+        highscore = [0]
+        file_path = os.path.join(get_activity_root(), 'data', 'highscore')
+        if os.path.exists(file_path):
+            with open(file_path, "r") as fp:
+                highscore = fp.readlines()
+        return int(highscore[0])
+
+    def save_highscore(self):
+        file_path = os.path.join(get_activity_root(), 'data', 'highscore')
+        int_highscore = self.read_highscore()
+        if not int_highscore > score:
+            with open(file_path, "w") as fp:
+                fp.write(str(score))
+
+    def load_highscore(self):
+        highscore = self.read_highscore()
+        try:
+            return highscore
+        except (ValueError, IndexError) as e:
+            logging.exception(e)
+            return 0
+        return 0
```

Comment:
I'm curious, what cases did you notice that made this threw those exceptions?

Reply:
Hmm, It might not be needed there now. There was a possibility back when a list was passed, later I typecasted it to `int` before passing it here and forgot to move the exception to `read_highscore` function. I'll fix it.

Reply:
Okay, thanks!